### PR TITLE
[BugFix] Be maybe crash when we do schema change for primary key table with separate primary key and sort key

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

If we create a primary key table with a separate primary key and sort key,  BE maybe crash if we do a schema change in the table(add column/drop column)
The reason is that we use column id as the sort key index in the tablet, and we only support reordering the sort key in this pr(https://github.com/StarRocks/starrocks/pull/13642) which assumes the number of columns in the origin tablet and new tablet during schema change is the same. However, if we add/drop a column, the sort key index in the new tablet will be different from the origin tablet and we will execute reorder which may cause BE crash.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/storage/tablet_manager.cpp [hunk 1]
```diff
@@ -1212,8 +1212,10 @@ Status TabletManager::_create_tablet_meta_unlocked(const TCreateTabletReq& reque
         next_unique_id = base_tablet->next_unique_id();
         size_t old_num_columns = base_tablet->num_columns();
         const auto& new_columns = request.tablet_schema.columns;
+        LOG(INFO) << "new columns size: " << new_columns.size();
```
Comment (chaoyli) [inline be/src/storage/tablet_manager.cpp:1215]: The log is useless.

File: be/src/storage/schema_change.cpp [hunk 2]
```diff
@@ -549,15 +549,25 @@ Status SchemaChangeHandler::_do_process_alter_tablet_v2(const TAlterTabletReqV2&
     if (base_tablet->keys_type() == KeysType::PRIMARY_KEYS) {
         const auto& base_sort_key_idxes = base_tablet->tablet_schema().sort_key_idxes();
         const auto& new_sort_key_idxes = new_tablet->tablet_schema().sort_key_idxes();
-        if (std::mismatch(new_sort_key_idxes.begin(), new_sort_key_idxes.end(), base_sort_key_idxes.begin()).first !=
-            new_sort_key_idxes.end()) {
+        std::vector<int32_t> base_sort_key_unique_ids;
```
Comment (chaoyli) [inline be/src/storage/schema_change.cpp:552]: It's better to describe the bug clearly.

Comment (sevev) [reply be/src/storage/schema_change.cpp:552]: > It's better to describe the bug clearly.

done, add more detailed description.

**Comment by decster** [comment]
```
Error:  COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
Error:  /home/runner/work/starrocks/starrocks/fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java:[331,25] cannot find symbol
```

**Comment by sevev** [comment]
> ```
> Error:  COMPILATION ERROR : 
> [INFO] -------------------------------------------------------------
> Error:  /home/runner/work/starrocks/starrocks/fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeJobV2.java:[331,25] cannot find symbol
> ```

done, fixed.

**Review by decster — DISMISSED** [review]

**Comment by sevev** [comment]
@chaoyli @decster PTAL~

**Review by chaoyli — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**